File: install_tools.js
const { execSync, exec } = require('child_process');
const os = require('os');

// 映射平台名称
function getPlatform() {
    const platform = os.platform();
    if (platform === 'win32') return 'windows';
    if (platform === 'darwin') return 'darwin';
    return platform; // linux 或其他
}

/**
 * 安装ADB工具
 * @returns {{success: boolean, message: string}} 安装结果
 */
function installAdb() {
    const platform = getPlatform();
    console.log("🔧 开始安装 ADB...");

    try {
        if (platform === 'windows') {
            execSync('winget install --id Google.PlatformTools --source winget', { stdio: 'inherit' });
            return { success: true, message: "✅ ADB 安装成功！" };
        } 
        else if (platform === 'darwin') {
            execSync('brew install android-platform-tools', { stdio: 'inherit' });
            return { success: true, message: "✅ ADB 安装成功！" };
        } 
        else if (platform === 'linux') {
            const packageManagers = [
                { cmd: 'apt', args: ['sudo', 'apt', 'install', '-y', 'android-tools-adb'] },
                { cmd: 'dnf', args: ['sudo', 'dnf', 'install', '-y', 'android-tools-adb'] },
                { cmd: 'pacman', args: ['sudo', 'pacman', '-S', '--noconfirm', 'android-tools'] },
                { cmd: 'zypper', args: ['sudo', 'zypper', 'install', '-y', 'android-tools'] }
            ];

            for (const pm of packageManagers) {
                try {
                    execSync(`which ${pm.cmd}`, { stdio: 'ignore' });
                    execSync(pm.args.join(' '), { stdio: 'inherit' });
                    return { success: true, message: "✅ ADB 安装成功！" };
                } catch (e) {
                    // 尝试下一个包管理器
                }
            }
            throw new Error("无法确定包管理器");
        } 
        else {
            throw new Error(`不支持的操作系统: ${platform}`);
        }
    } catch (error) {
        if (platform === 'windows') {
            return { 
                success: false, 
                message: `❌ ADB 安装失败: ${error.message}\n请确保已安装 winget: https://aka.ms/getwinget`
            };
        } 
        else if (platform === 'darwin') {
            return { 
                success: false, 
                message: `❌ ADB 安装失败: ${error.message}\n请确保已安装 Homebrew: https://brew.sh/`
            };
        } 
        else if (platform === 'linux') {
            return {
                success: false,
                message: '❌ 无法确定包管理器\n请手动安装: Debian/Ubuntu: sudo apt install android-tools-adb\n' +
                         'Fedora/RHEL: sudo dnf install android-tools-adb\n' +
                         'Arch/Manjaro: sudo pacman -S android-tools\n' +
                         'openSUSE: sudo zypper install android-tools'
            };
        } 
        else {
            return { success: false, message: `❌ 不支持的操作系统: ${platform}` };
        }
    }
}

/**
 * 安装FFmpeg工具
 * @returns {{success: boolean, message: string}} 安装结果
 */
function installFfmpeg() {
    const platform = getPlatform();
    console.log("🔧 开始安装 FFmpeg...");

    try {
        if (platform === 'windows') {
            execSync('winget install --id Gyan.FFmpeg --source winget', { stdio: 'inherit' });
            return { success: true, message: "✅ FFmpeg 安装成功！" };
        } 
        else if (platform === 'darwin') {
            execSync('brew install ffmpeg', { stdio: 'inherit' });
            return { success: true, message: "✅ FFmpeg 安装成功！" };
        } 
        else if (platform === 'linux') {
            const packageManagers = [
                { cmd: 'apt', args: ['sudo', 'apt', 'install', '-y', 'ffmpeg'] },
                { cmd: 'dnf', args: ['sudo', 'dnf', 'install', '-y', 'ffmpeg'] },
                { cmd: 'pacman', args: ['sudo', 'pacman', '-S', '--noconfirm', 'ffmpeg'] },
                { cmd: 'zypper', args: ['sudo', 'zypper', 'install', '-y', 'ffmpeg'] }
            ];

            for (const pm of packageManagers) {
                try {
                    execSync(`which ${pm.cmd}`, { stdio: 'ignore' });
                    execSync(pm.args.join(' '), { stdio: 'inherit' });
                    return { success: true, message: "✅ FFmpeg 安装成功！" };
                } catch (e) {
                    // 尝试下一个包管理器
                }
            }
            throw new Error("无法确定包管理器");
        } 
        else {
            throw new Error(`不支持的操作系统: ${platform}`);
        }
    } catch (error) {
        if (platform === 'windows') {
            return { 
                success: false, 
                message: `❌ FFmpeg 安装失败: ${error.message}\n请确保已安装 winget: https://aka.ms/getwinget`
            };
        } 
        else if (platform === 'darwin') {
            return { 
                success: false, 
                message: `❌ FFmpeg 安装失败: ${error.message}\n请确保已安装 Homebrew: https://brew.sh/`
            };
        } 
        else if (platform === 'linux') {
            return {
                success: false,
                message: '❌ 无法确定包管理器\n请手动安装: Debian/Ubuntu: sudo apt install ffmpeg\n' +
                         'Fedora/RHEL: sudo dnf install ffmpeg\n' +
                         'Arch/Manjaro: sudo pacman -S ffmpeg\n' +
                         'openSUSE: sudo zypper install ffmpeg'
            };
        } 
        else {
            return { success: false, message: `❌ 不支持的操作系统: ${platform}` };
        }
    }
}

/**
 * 验证工具是否安装成功
 * @param {'adb' | 'ffmpeg'} toolName 工具名称
 * @returns {{success: boolean, message: string}} 验证结果
 */
function verifyInstallation(toolName) {
    try {
        if (toolName === 'adb') {
            const output = execSync('adb --version', { encoding: 'utf-8' });
            const versionLine = output.split('\n')[0];
            return { success: true, message: `✅ ADB 验证成功: ${versionLine}` };
        } 
        else if (toolName === 'ffmpeg') {
            const output = execSync('ffmpeg -version', { encoding: 'utf-8' });
            const version = output.split('\n')[0].split(' ')[2];
            return { success: true, message: `✅ FFmpeg 验证成功: 版本 ${version}` };
        }
    } catch (error) {
        return { success: false, message: `⚠️ 找不到 ${toolName} 命令，请尝试重新启动终端或手动添加环境变量` };
    }
}

/**
 * 安装所有必要工具（ADB和FFmpeg）
 * @returns {Object} 安装结果
 */
function installAll() {
    const results = {
        adb: { success: false, message: '' },
        ffmpeg: { success: false, message: '' },
        adb_verify: { success: false, message: '' },
        ffmpeg_verify: { success: false, message: '' }
    };

    // 安装ADB
    const adbResult = installAdb();
    results.adb = adbResult;

    // 安装FFmpeg
    const ffmpegResult = installFfmpeg();
    results.ffmpeg = ffmpegResult;

    // 验证安装
    if (adbResult.success) {
        results.adb_verify = verifyInstallation('adb');
    }
    if (ffmpegResult.success) {
        results.ffmpeg_verify = verifyInstallation('ffmpeg');
    }

    return results;
}

// 主程序入口
if (require.main === module) {
    console.log("=".repeat(50));
    console.log(`🚀 正在为 ${os.platform()} 安装必要工具...`);
    console.log("=".repeat(50));

    const results = installAll();

    console.log("\n📊 安装结果摘要:");
    console.log(`ADB 安装: ${results.adb.success ? '成功' : '失败'} - ${results.adb.message}`);
    console.log(`ADB 验证: ${results.adb_verify.success ? '成功' : '失败'} - ${results.adb_verify.message}`);
    console.log(`FFmpeg 安装: ${results.ffmpeg.success ? '成功' : '失败'} - ${results.ffmpeg.message}`);
    console.log(`FFmpeg 验证: ${results.ffmpeg_verify.success ? '成功' : '失败'} - ${results.ffmpeg_verify.message}`);

    console.log("\n" + "=".repeat(50));
    console.log("💡 提示: 如果验证失败，请尝试重启终端后再试");
    console.log("=".repeat(50));
}

// 暴露接口
module.exports = {
    installAdb,
    installFfmpeg,
    verifyInstallation,
    installAll
};

File: install_tools.py
import os
import sys
import subprocess
import platform

def install_adb():
    """
    安装ADB工具
    返回: (success, message) 元组，success为布尔值表示是否成功
    """
    system = platform.system().lower()
    print(f"🔧 开始安装 ADB...")
    
    if system == "windows":
        try:
            result = subprocess.run(
                ["winget", "install", "--id", "Google.PlatformTools", "--source", "winget"],
                check=True,
                capture_output=True,
                text=True
            )
            return True, "✅ ADB 安装成功！"
        except subprocess.CalledProcessError as e:
            return False, f"❌ ADB 安装失败: {e.stderr}\n请确保已安装 winget: https://aka.ms/getwinget"
        except FileNotFoundError:
            return False, "❌ 未找到 winget 命令\n请安装 Windows Package Manager: https://aka.ms/getwinget"

    elif system == "darwin":  # macOS
        try:
            result = subprocess.run(
                ["brew", "install", "android-platform-tools"],
                check=True,
                capture_output=True,
                text=True
            )
            return True, "✅ ADB 安装成功！"
        except subprocess.CalledProcessError as e:
            return False, f"❌ ADB 安装失败: {e.stderr}\n请确保已安装 Homebrew: https://brew.sh/"
        except FileNotFoundError:
            return False, "❌ 未找到 brew 命令\n请先安装 Homebrew: https://brew.sh/"

    elif system == "linux":
        managers = {
            "apt": ["sudo", "apt", "install", "-y", "android-tools-adb"],
            "dnf": ["sudo", "dnf", "install", "-y", "android-tools-adb"],
            "pacman": ["sudo", "pacman", "-S", "--noconfirm", "android-tools"],
            "zypper": ["sudo", "zypper", "install", "-y", "android-tools"]
        }
        
        for cmd, install_cmd in managers.items():
            if subprocess.run(["which", cmd], capture_output=True).returncode == 0:
                try:
                    subprocess.run(install_cmd, check=True, capture_output=True)
                    return True, "✅ ADB 安装成功！"
                except subprocess.CalledProcessError as e:
                    return False, f"❌ 使用 {cmd} 安装失败: {e.stderr}"
        
        return False, "❌ 无法确定包管理器\n请手动安装: Debian/Ubuntu: sudo apt install android-tools-adb\nFedora/RHEL: sudo dnf install android-tools-adb\nArch/Manjaro: sudo pacman -S android-tools\nopenSUSE: sudo zypper install android-tools"

    else:
        return False, f"❌ 不支持的操作系统: {system}"

def install_ffmpeg():
    """
    安装FFmpeg工具
    返回: (success, message) 元组，success为布尔值表示是否成功
    """
    system = platform.system().lower()
    print(f"🔧 开始安装 FFmpeg...")
    
    if system == "windows":
        try:
            result = subprocess.run(
                ["winget", "install", "--id", "Gyan.FFmpeg", "--source", "winget"],
                check=True,
                capture_output=True,
                text=True
            )
            return True, "✅ FFmpeg 安装成功！"
        except subprocess.CalledProcessError as e:
            return False, f"❌ FFmpeg 安装失败: {e.stderr}\n请确保已安装 winget: https://aka.ms/getwinget"
        except FileNotFoundError:
            return False, "❌ 未找到 winget 命令\n请安装 Windows Package Manager: https://aka.ms/getwinget"

    elif system == "darwin":  # macOS
        try:
            result = subprocess.run(
                ["brew", "install", "ffmpeg"],
                check=True,
                capture_output=True,
                text=True
            )
            return True, "✅ FFmpeg 安装成功！"
        except subprocess.CalledProcessError as e:
            return False, f"❌ FFmpeg 安装失败: {e.stderr}\n请确保已安装 Homebrew: https://brew.sh/"
        except FileNotFoundError:
            return False, "❌ 未找到 brew 命令\n请先安装 Homebrew: https://brew.sh/"

    elif system == "linux":
        managers = {
            "apt": ["sudo", "apt", "install", "-y", "ffmpeg"],
            "dnf": ["sudo", "dnf", "install", "-y", "ffmpeg"],
            "pacman": ["sudo", "pacman", "-S", "--noconfirm", "ffmpeg"],
            "zypper": ["sudo", "zypper", "install", "-y", "ffmpeg"]
        }
        
        for cmd, install_cmd in managers.items():
            if subprocess.run(["which", cmd], capture_output=True).returncode == 0:
                try:
                    subprocess.run(install_cmd, check=True, capture_output=True)
                    return True, "✅ FFmpeg 安装成功！"
                except subprocess.CalledProcessError as e:
                    return False, f"❌ 使用 {cmd} 安装失败: {e.stderr}"
        
        return False, "❌ 无法确定包管理器\n请手动安装: Debian/Ubuntu: sudo apt install ffmpeg\nFedora/RHEL: sudo dnf install ffmpeg\nArch/Manjaro: sudo pacman -S ffmpeg\nopenSUSE: sudo zypper install ffmpeg"

    else:
        return False, f"❌ 不支持的操作系统: {system}"

def verify_installation(tool_name):
    """
    验证工具是否安装成功
    :param tool_name: 'adb' 或 'ffmpeg'
    :return: (success, version_info) 元组
    """
    try:
        if tool_name == "adb":
            result = subprocess.run(
                ["adb", "--version"],
                capture_output=True,
                text=True
            )
            if result.returncode == 0:
                version_line = result.stdout.splitlines()[0]
                return True, f"✅ ADB 验证成功: {version_line}"
        elif tool_name == "ffmpeg":
            result = subprocess.run(
                ["ffmpeg", "-version"],
                capture_output=True,
                text=True
            )
            if result.returncode == 0:
                version_line = result.stdout.splitlines()[0].split()[2]
                return True, f"✅ FFmpeg 验证成功: 版本 {version_line}"
    except:
        pass
    
    return False, f"⚠️ 找不到 {tool_name} 命令，请尝试重新启动终端或手动添加环境变量"

def install_all():
    """
    安装所有必要工具（ADB和FFmpeg）
    返回: 包含安装结果的字典
    """
    results = {
        "adb": {"success": False, "message": ""},
        "ffmpeg": {"success": False, "message": ""},
        "adb_verify": {"success": False, "message": ""},
        "ffmpeg_verify": {"success": False, "message": ""}
    }
    
    # 安装ADB
    adb_success, adb_msg = install_adb()
    results["adb"]["success"] = adb_success
    results["adb"]["message"] = adb_msg
    
    # 安装FFmpeg
    ffmpeg_success, ffmpeg_msg = install_ffmpeg()
    results["ffmpeg"]["success"] = ffmpeg_success
    results["ffmpeg"]["message"] = ffmpeg_msg
    
    # 验证安装
    if adb_success:
        verify_success, verify_msg = verify_installation("adb")
        results["adb_verify"]["success"] = verify_success
        results["adb_verify"]["message"] = verify_msg
    
    if ffmpeg_success:
        verify_success, verify_msg = verify_installation("ffmpeg")
        results["ffmpeg_verify"]["success"] = verify_success
        results["ffmpeg_verify"]["message"] = verify_msg
    
    return results

if __name__ == "__main__":
    print("="*50)
    print(f"🚀 正在为 {platform.system()} 安装必要工具...")
    print("="*50)
    
    results = install_all()
    
    print("\n📊 安装结果摘要:")
    print(f"ADB 安装: {'成功' if results['adb']['success'] else '失败'} - {results['adb']['message']}")
    print(f"ADB 验证: {'成功' if results['adb_verify']['success'] else '失败'} - {results['adb_verify']['message']}")
    print(f"FFmpeg 安装: {'成功' if results['ffmpeg']['success'] else '失败'} - {results['ffmpeg']['message']}")
    print(f"FFmpeg 验证: {'成功' if results['ffmpeg_verify']['success'] else '失败'} - {results['ffmpeg_verify']['message']}")
    
    print("\n" + "="*50)
    print("💡 提示: 如果验证失败，请尝试重启终端后再试")
    print("="*50)